### PR TITLE
helm: Add loki persistence option

### PR DESCRIPTION
**Pull request by rlaisqls**
**What this PR does / why we need it**:

For users who want to install and run loki using empty dir volume rather than PVC, option is needed to choose whether to persist.

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
  - [x] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: production/helm/loki/values.yaml [hunk 1]
```diff
@@ -793,6 +793,8 @@ write:
   # -- The default is to deploy all pods in parallel.
   podManagementPolicy: "Parallel"
   persistence:
+    # -- Enable data persistence using PVC(s)
```
Comment (JStickler) [inline production/helm/loki/values.yaml:796]:
```suggestion
    # -- Enable data persistence using persistent volume claims (PVCs)
```

File: production/helm/loki/values.yaml [hunk 2]
```diff
@@ -964,6 +966,8 @@ read:
...
```
Comment (JStickler) [inline production/helm/loki/values.yaml:969]:
```suggestion
    # -- Enable data persistence using persistent volume claims (PVCs)
```

File: production/helm/loki/values.yaml [hunk 3]
```diff
@@ -1065,6 +1069,8 @@ backend:
...
```
Comment (JStickler) [inline production/helm/loki/values.yaml:1072]:
```suggestion
    # -- Enable data persistence using persistent volume claims (PVCs)
```

File: docs/sources/setup/install/helm/reference.md [hunk 4]
```diff
@@ -202,6 +202,15 @@ null
 			<td><pre lang="json">
 {}
 </pre>
+</td>
+		</tr>
+		<tr>
+			<td>backend.persistence.enable</td>
+			<td>bool</td>
+			<td>Enable data persistence using PVC(s)</td>
```
Comment (JStickler) [inline docs/sources/setup/install/helm/reference.md:210]:
```suggestion
			<td>Enable data persistence using persistent volume claims (PVCs)</td>
```

File: docs/sources/setup/install/helm/reference.md [hunk 5]
```diff
@@ -3256,6 +3265,15 @@ false
 			<td><pre lang="json">
 {}
 </pre>
+</td>
+		</tr>
+		<tr>
+			<td>read.persistence.enable</td>
+			<td>bool</td>
+			<td>Enable data persistence using PVC(s)</td>
```
Comment (JStickler) [inline docs/sources/setup/install/helm/reference.md:3273]:
```suggestion
			<td>Enable data persistence using persistent volume claims (PVCs)</td>
```

File: docs/sources/setup/install/helm/reference.md [hunk 6]
```diff
@@ -4536,6 +4554,15 @@ null
 			<td><pre lang="json">
 {}
 </pre>
+</td>
+		</tr>
+		<tr>
+			<td>write.persistence.enable</td>
+			<td>bool</td>
+			<td>Enable data persistence using PVC(s)</td>
```
Comment (JStickler) [inline docs/sources/setup/install/helm/reference.md:4562]:
```suggestion
			<td>Enable data persistence using persistent volume claims (PVCs)</td>
```

**Review by JStickler — COMMENTED** [review]
[docs team]

**Review by JStickler — COMMENTED** [review]
[Docs team]  Docs LGTM, someone from engineering will still need to review and approve.

**Comment by MichelHollands** [comment]
@rlaisqls Can you use the version 5.26.0? It has to be updated in all 3 places production/helm/loki/Chart.yaml,  production/helm/loki/CHANGELOG.md and production/helm/loki/README.md. Once that is done I'll merge it.

**Comment by MichelHollands** [comment]
@rlaisqls My apologies. There is another earlier PR that also updates the persistence option: https://github.com/grafana/loki/pull/10338. Would that work instead of this PR?

**Comment by rlaisqls** [comment]
@MichelHollands Because the current version in the branch is 2.24.0, I raised the version to 2.25.0.
You can merge the PR you linked with instead of my PR, but the persist option for the 'read' component is missing there. I think it would be nice to merge this PR to prevent further demands on the persiste option, or add that code to that PR. Please do whatever you want. 🙂

**Comment by matschaffer-roblox** [comment]
This PR should serve my need as well. I skipped read in my PR because in my case the chart uses https://github.com/grafana/loki/blob/main/production/helm/loki/templates/read/deployment-read.yaml rather than https://github.com/grafana/loki/blob/main/production/helm/loki/templates/read/statefulset-read.yaml

**Comment by rlaisqls** [comment]
I see. You're right. I'll close this PR :)

**Comment by matschaffer-roblox** [comment]
cool, thanks @rlaisqls - I'll update mine to be ready for merge.

**Pull request closed without merging**